	- pass CFLAGS to compiler
	- pass $(M) when compiling deadwood, no idea why that is missing originally
	- do not pass $(M) when compiling in tcp/, as that would cause problems with
	  the implicit rules in there instead, pass CFLAGS explicitly otherwise
	  they're lost for some reason
	- do not use V as variable name, conflicts with one from OpenADK
--- maradns-2.0.09.orig/build/Makefile.linux	2010-08-31 05:56:38.000000000 +0200
+++ maradns-2.0.09/build/Makefile.linux	2014-06-19 03:16:54.570689574 +0200
@@ -23,33 +23,33 @@ EXECS=server/maradns
 # LDFLAGS=-L/usr/ucblib -lucb -lxnet
 # end the Solaris section
 # Non-Solaris version of "M"
-V="VERSION=$(VERSION)"
+VER="VERSION=$(VERSION)"
 Q="DEFINES=-DSELECT_PROBLEM"
 
 # Debug
 
 FLAGS = -O2 -Wall -DSELECT_PROBLEM
-M="CC=$(CC) $(FLAGS)"
-D="CC=$(CC) $(FLAGS) -DDEBUG -DTHREADS"
+M="CC=$(CC) $(CFLAGS) $(FLAGS)"
+D="CC=$(CC) $(CFLAGS) $(FLAGS) -DDEBUG -DTHREADS"
 #FLAGS = -g
 
 all: 
 	cd libs ; make $(M) ; cd ../dns ; make $(M) ; \
 	cd ../rng ; make $(M) ; cd ../parse ; make $(M) ; \
 	cd ../qual ; make $(M) ; cd ../server ; \
-	make $(M) $(V) COMPILED=\"$(COMPILED)\" ; \
+	make $(M) $(VER) COMPILED=\"$(COMPILED)\" ; \
 	cd ../tools ; make $(M) ; \
-	cd ../deadwood-*/src/ ; make FLAGS=-O2 ; \
-	cd ../../tcp ; make $(M) $(V) ; cat ../00README.FIRST
+	cd ../deadwood-*/src/ ; make FLAGS=-O2 $(M) ; \
+	cd ../../tcp ; make CFLAGS="$(CFLAGS)" $(VER) ; cat ../00README.FIRST
 
 debug: 
 	cd libs ; make $(D) DEBUG="-DDEBUG -DTHREADS" ; \
 	cd ../dns ; make $(D) ; cd ../rng ; make $(D) ; \
 	cd ../parse ; make $(D) ; cd ../qual ; make $(D) ; \
 	cd ../server ; \
-	make $(D) $(Q) $(V) COMPILED=\"$(COMPILED_DEBUG)\" ; \
+	make $(D) $(Q) $(VER) COMPILED=\"$(COMPILED_DEBUG)\" ; \
 	cd ../tools ; make $(D) ; \
-	cd ../tcp ; make $(D) $(V) ; cat ../00README.FIRST
+	cd ../tcp ; make $(D) $(VER) ; cat ../00README.FIRST
 
 clean:
 	rm -f $(OBJECTS) core $(EXECS) ; \
